core/server/src/main/java/alluxio/worker/file/UnderFileSystemManager.java

```diff
@@ -193,7 +194,10 @@ private InputStream openAtPosition(long position) throws IOException {
         }
         UnderFileSystem ufs = UnderFileSystem.get(mUri);
         // TODO(calvin): Consider making openAtPosition part of the UFS API
-        if (ufs instanceof S3UnderFileSystem) { // Optimization for S3 UFS
+        if (ufs instanceof S3AUnderFileSystem) { // Optimization for S3A UFS
+          mStream =
+              new CountingInputStream(((S3AUnderFileSystem) ufs).openAtPosition(mUri, position));
```

Comment:
Good catch, it's definitely needed. Unfortunately due to the hardcoding here it was not caught in any unit tests. I've added a test for it but it is far from ideal (ie. only tests one of the 3 ufses).